### PR TITLE
add support for using a custom context-path

### PR DESCRIPTION
**Pull request by duoduobingbing**
Added support for auto inferring the context path if one is set via `server.servlet.context-path` from the Spring Boot properties.

Introduced new setting `spring.dashboard.openUrl` to control with which path the Spring Boot app should be opened. 
`null` is default behavior and resolves to `http://localhost:{port}{contextPath}`.

Closes https://github.com/microsoft/vscode-spring-boot-dashboard/issues/161

### CODE REVIEW

File: java-extension/src/main/java/com/microsoft/vscode/spring/boot/dashboard/java/extension/JmxService.java [hunk 1]
```diff
@@ -12,15 +14,42 @@
...
+
+		}catch(InstanceNotFoundException ex){
+			return null;
+		}
+	}
+
 	static String getPortViaAdmin(MBeanServerConnection connection) throws Exception {
 		try {
 			String DEFAULT_OBJECT_NAME = "org.springframework.boot:type=Admin,name=SpringApplication";
 			ObjectName objectName = new ObjectName(DEFAULT_OBJECT_NAME);
```
Comment (Eskibear) [inline java-extension/src/main/java/com/microsoft/vscode/spring/boot/dashboard/java/extension/JmxService.java:41]: can reuse `ADMIN_DEFAULT_OBJECT_NAME`

Comment (Eskibear) [reply java-extension/src/main/java/com/microsoft/vscode/spring/boot/dashboard/java/extension/JmxService.java:41]: or remove this method if not used any more.

**Review by Eskibear — COMMENTED** [review]
- This PR changes how extension parse output from java process. 
  - If so, I prefer to use a more extensible and reliable way, e.g. using json format.

- You added a new setting specifying url format. 
  - I don't think we need this setting as we can always use `{host}:{port}{context-path}` .

File: java-extension/src/main/java/com/microsoft/vscode/spring/boot/dashboard/java/extension/JmxService.java (hunk 1 above)
Comment (duoduobingbing) [reply java-extension/src/main/java/com/microsoft/vscode/spring/boot/dashboard/java/extension/JmxService.java]: I forgot to remove the original `getPortViaAdmin` method

**Comment by duoduobingbing** [comment]
Thank you for looking into this PR so quickly.

> You added a new setting specifying url format.
> I don't think we need this setting as we can always use {host}:{port}{context-path} .

The setting is very useful for three use cases:
* If the Spring app has a different entry point than `{host}:{port}{context-path}` e.g. `http://localhost:8080/example/coolForm.jsp`
* For switching to `https`. With the current JMX Spring Admin Bean setup there is no easy way to get the `{host}` placeholder resolved dynamically, so for now this property should suffice to archive that.
* If the context path is defined via `WebServerFactoryCustomizer` it cannot be correctly inferred from `server.servlet.context-path`. 

Thus, I think it is beneficial to have a property to control at which URL the application is opened

**Comment by Eskibear** [comment]
I'll take a look and merge it if there's no problem.  And just FYI, in long term plan, we might refactor to fetch information from spring-tools extension.

**Comment by duoduobingbing** [comment]
> I'll take a look and merge it if there's no problem. And just FYI, in long term plan, we might refactor to fetch information from spring-tools extension.

Is there any estimate as to approximately when this will be published on the VSCode marketplace (assuming that there is no problem with the code I wrote)?

**Comment by Eskibear** [comment]
Usually it's a monthly release. You can expect this in the end of this month. BTW we also have nightly build in pre-release channel, you can switch to pre-release version in your vscode.

**Review by Eskibear — APPROVED** [review]

**Pull request merged**